File: 2015/day04.py
import hashlib

string = "ckczppom"
start = 1

while True:
    puzzle = string + str(start)
    result = hashlib.md5(bytes(puzzle, 'utf-8'))
    digest = result.hexdigest()
    if digest[0:5] == "00000":
        print(digest)
        print(start)
    if digest[0:6] == "000000":
        print(digest)
        print(start)
        break
    start += 1

File: 2015/day04.rb
require 'digest'

string = "ckczppom"
start = 1

while true do
    puzzle = string + start.to_s
    digest = Digest::MD5.hexdigest(puzzle)
    if digest.to_s[0...5] == "00000"
        puts(digest)
        puts(start)
    end
    if digest.to_s[0...6] == "000000"
        puts(digest)
        puts(start)
        exit
    end
    start += 1
end
